src/api/spec/models/event_find_subscriptions_spec.rb

```diff
@@ -1,37 +1,139 @@
-
-    let!(:comment) { create(:comment_project, commentable: project, body: "Hey @#{user1.login} how are things?", user: comment_author) }
-    let(:event) { Event::CommentForProject.first }
-
-    subject! { EventFindSubscriptions.new(event).subscriptions }
-
-    it 'includes the users and groups subscribed to Event::CommentForProject' do
-      expect(subject).to include(subscription2, subscription3, subscription4, subscription5)
+    subject do
+      event = Event::CommentForProject.first
+      EventFindSubscriptions.new(event).subscriptions
     end
 
-    it 'does not include the author of the comment' do
-      expect(subject).not_to include(subscription1)
-    end
+    context 'with a comment for a project' do
+      context 'with no maintainers' do
+        let!(:project) { create(:project) }
+        let!(:comment) { create(:comment_project, commentable: project) }
+
+        it 'does not include the author of the comment' do
+          expect(subject.map(&:subscriber)).not_to include(comment.user)
+        end
+      end
+
+      it_behaves_like 'it returns subscriptions for an event' do
+        let!(:subscriber) { create(:confirmed_user) }
+      end
+
+      it_behaves_like 'it returns subscriptions for an event' do
+        let!(:subscriber) { create(:group) }
+      end
+
+      context 'with a maintainer group who has no email set and has a user as a member' do
+        let!(:group) { create(:group, email: nil) }
+        let!(:user) { create(:confirmed_user) }
+
+        let!(:project) { create(:project, maintainer: [group]) }
+        let!(:comment) { create(:comment_project, commentable: project) }
+
+        before do
+          group.users << user
+        end
+
+        context 'and the user has a maintainer subscription' do
+          context 'which is enabled' do
+            let!(:subscription) { create(:event_subscription_comment_for_project_no_subscriber, receiver_role: 'maintainer', subscriber: user) }
+
+            it 'returns the subscription for that user/group' do
+              subscriber_result = subject.find { |subscription| subscription.subscriber == user }
+
+              expect(subscriber_result).to eq(subscription)
+            end
+          end
+
+          context 'which is disabled' do
+            let!(:subscription) { create(:event_subscription_comment_for_project_no_subscriber, receiver_role: 'maintainer', subscriber: user, channel: 'disabled') }
+
+            it 'does not include that user/group' do
+              expect(subject.map(&:subscriber)).not_to include(user)
+            end
+          end
+        end
+
+        context 'and the user has no subscriptions' do
+          context 'and a global maintainer subscription is enabled' do
+            let!(:global_subscription) do
+              create(:event_subscription_comment_for_project, receiver_role: 'maintainer', user: nil, group: nil)
+            end
+
+            # TODO: This does not seem like the correct logic for this class.
```

Comment:
@coolo maybe you have an idea about this? Currently if a user belongs to a group which is a maintainer of a project then that user will only receive notifications about events for the project if they have specifically set that they want to receive maintainer notifications. So the default setting might be that maintainers receive notifications, but this guy won't. That doesn't seem to match up with the way the event notifications work everywhere else?